connector/src/main/scala/com/basho/riak/spark/writer/mapper/SqlDataMapper.scala

```diff
@@ -0,0 +1,78 @@
+package com.basho.riak.spark.writer.mapper
+
+import java.util.{Calendar, Date}
+
+import com.basho.riak.client.core.query.timeseries.ColumnDescription.ColumnType
+import com.basho.riak.client.core.query.timeseries.{Cell, ColumnDescription, Row => RiakRow}
+import com.basho.riak.spark.rdd.BucketDef
+import com.basho.riak.spark.writer.ts.RowDef
+import com.basho.riak.spark.writer.{WriteDataMapper, WriteDataMapperFactory}
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.{Row => SparkRow}
+
+/**
+  * [[WriteDataMapper]] that converts Spark [[SparkRow]] objects to [[RiakRow]]
+  */
+class SqlDataMapper[T <: SparkRow] extends WriteDataMapper[T, RowDef] {
+
+  override def mapValue(row: T): RowDef = {
+    Option(row.schema) match {
+      case None => RowDef(createRowByType(row), None)
+      case Some(schema) =>
+        val (riakRow, columnsDef) = createRowBySchema(row, schema)
+        RowDef(riakRow, columnsDef)
+    }
+  }
+
+  private def createRowBySchema(row: T, schema: StructType): (RiakRow, Option[Seq[ColumnDescription]]) = {
+    val cellsWithDescription = schema.fields.map(field => getCellAndDescription(row, field))
+    val (descriptions, cells) = cellsWithDescription.unzip
+    new RiakRow(cells: _*) -> Option(descriptions)
+  }
+
+  private def getCellAndDescription(row: T, field: StructField): (ColumnDescription, Cell) = field.dataType match {
```

Comment:
In case of null value new Cell(null) is returned. Should return just null object instead of Cell.